dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/icon/IconTest.java

```diff
@@ -102,6 +104,26 @@ void shouldGetIconDataWhenKeyBelongsToIcon() throws NotFoundException {
     assertNotNull(iconService.getIconResource(Key));
   }
 
+  @Test
+  void shouldGetIconsMatchingILikeKey()
+      throws SQLException, BadRequestException, NotFoundException {
+    Icon icon1 = createIcon('I', keywords, createAndPersistFileResource('I'));
+    Icon icon2 = createIcon('J', keywords, createAndPersistFileResource('J'));
+
+    icon1.setKey("lab-agent");
+    icon2.setKey("facility-agent");
+
+    iconService.addIcon(icon1);
+    iconService.addIcon(icon2);
+
+    IconOperationParams operationParams = new IconOperationParams();
+    operationParams.setSearch("agent");
+    List<Icon> icons = iconService.getIcons(operationParams);
+
+    assertThat(icons.size(), is(2));
```

Comment:
if order matters please use `assertEquals(List.of(), actual)` else use our helper `assertContainsOnly` from org.hisp.dhis.utils.Assertions. See https://github.com/dhis2/wow-backend/blob/59d3a443c755080a0da845251ca5db2d07dcd4a0/guides/testing/testing.md?plain=1#L109

Reply:
Please remove `assertThat(icons.size(), is(2));` as this will already be caught by `assertContainsOnly`. The way it is right now the test would fail with `expected 2 got 3` or so. We want to avoid such assertion errors in CI as it does not tell you enough to find the issue. That is why we use `assertContainsOnly` or `assertEquals(List.of(), actual)`. These two already assert on the size but print what's in the collection in case the assertion fails.